### PR TITLE
Coupon

### PR DESCRIPTION
**Pull request by Mary0110**

### CODE REVIEW

File: ModsenOnlineStore.Store.API/Controllers/CouponsController.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        [HttpGet]
+        public async Task<IActionResult> GetAllCoupons() =>
+            Ok(await couponService.GetAllCoupons());
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetCoupon(int id)
+        {
+            var couponInfo = await couponService.GetCoupon(id);
+
+            if (couponInfo.Success is false)
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.API/Controllers/CouponsController.cs]: you can write just `if (!couponInfo.Success)`

File: ModsenOnlineStore.Store.Domain/DTOs/CouponDTO/ApplyCouponDTO.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace ModsenOnlineStore.Store.Domain.DTOs.CouponDTO;
+
+public class ApplyCouponDTO
+{
+    public int couponId { get; set; }
+    
+    public int orderId { get; set; }
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Domain/DTOs/CouponDTO/ApplyCouponDTO.cs]: use PascalCase

File: ModsenOnlineStore.Store.Infrastructure/Data/CouponRepository.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            .Where(c => c.UserId == userId)
+            .ToListAsync();
+
+        context.Coupons.RemoveRange(_couponsToDelete);
+        await context.SaveChangesAsync();
+        
+        return _couponsToDelete;
+    }
+
+    public async Task<Order?> ApplyCoupon(int couponId, int orderId)
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Infrastructure/Data/CouponRepository.cs]: Why is this method in the repository?
repositories are used for data storage access only, all business rules should be in application layer, e.g. services

File: ModsenOnlineStore.Store.API/Controllers/CouponsController.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                return NotFound();
+            }
+
+            return Ok(couponInfo);
+        }
+
+        [HttpPost("apply")]
+        public async Task<IActionResult> ApplyCoupon(ApplyCouponDTO data)
+        {
+            return Ok(await couponService.ApplyCoupon(data));
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.API/Controllers/CouponsController.cs]: Is it really always Ok regardless ApplyCoupon operation result?

**Review by elv1sabene — CHANGES_REQUESTED** [review]

File: ModsenOnlineStore.Store.API/Controllers/CouponsController.cs (hunk 1 above)
Comment (Mary0110) [reply ModsenOnlineStore.Store.API/Controllers/CouponsController.cs]: fixed

File: ModsenOnlineStore.Store.Domain/DTOs/CouponDTO/ApplyCouponDTO.cs (hunk 2 above)
Comment (Mary0110) [reply ModsenOnlineStore.Store.Domain/DTOs/CouponDTO/ApplyCouponDTO.cs]: fixed

File: ModsenOnlineStore.Store.Infrastructure/Data/CouponRepository.cs (hunk 3 above)
Comment (Mary0110) [reply ModsenOnlineStore.Store.Infrastructure/Data/CouponRepository.cs]: fixed

File: ModsenOnlineStore.Store.API/Controllers/CouponsController.cs (hunk 4 above)
Comment (Mary0110) [reply ModsenOnlineStore.Store.API/Controllers/CouponsController.cs]: fixed

File: ModsenOnlineStore.Store.Infrastructure/Data/DbInitializer.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+namespace ModsenOnlineStore.Store.Infrastructure.Data;
+
+public class DbInitializer
+{
+    public static async Task SeedData(DataContext context)
+    {
+        var firstUser = new User()
+        {
+        };
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Infrastructure/Data/DbInitializer.cs]: do you need these braces?

File: ModsenOnlineStore.Store.Infrastructure/Data/DbInitializer.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+        var Ivan = new User()
+        {
+
+        };
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Infrastructure/Data/DbInitializer.cs]: same

File: ModsenOnlineStore.Store.Infrastructure/Data/DbInitializer.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                UserId = firstUser.Id
+            };
+            var secondc = new Domain.Entities.Coupon()
+            {
+                User = firstUser,
+                Discount = 15,
+                UserId = firstUser.Id
+            };
+
+            context.Coupons.AddRange(new Domain.Entities.Coupon[] { firstc, secondc });
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Infrastructure/Data/DbInitializer.cs]: why is there an indentation here?

File: ModsenOnlineStore.Store.Infrastructure/Data/DbInitializer.cs [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            };
+            var secondO = new Domain.Entities.Order()
+            {
+                User = Ivan,
+            };
+
+            context.Orders.AddRange(new Domain.Entities.Order[] { firstOrder, secondO });
+        }
+        
+        // if (!context.Coupons.Any()) {
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Infrastructure/Data/DbInitializer.cs]: do you need this?

**Review by elv1sabene — APPROVED** [review]

**Pull request merged**